main.cpp

```diff
@@ -1,25 +1,75 @@
 #include <iostream>
+#include <getopt.h>
 #include "common.h"
-#include "md5-duplicate-detection.h"
 #include "encrypt-decrypt.h"
 #include "files.h"
 
-int main() {
-    std::cout << "Hello, World!" << std::endl;
-    /*
-    std::string directory = "some directory";
+int main(int argc, char *argv[]) {
+
+    int opt, option_index = 0, eord = -1;
+    const char *directory;
     std::vector<fs::path> files;
-    getFiles(directory, files);
-    for (fs::path i: files)
-        std::cout << i.string() << ' ';
-        */
-
-//    fs::path A("test-files/A");
-//    fs::path B("test-files/B");
-//    is_data_duplicate(A,B);
-
-//    fs::path input("test-files/file.txt");
-//    encrypt(input);
-//    decrypt(input);
+    static struct option long_options[] =
+            {
+                    {"help",     no_argument, NULL,  'h' },
+                    {"encrypt",  no_argument, NULL,  'e' },
+                    {"decrypt",  no_argument, NULL,  'd' },
+                    {NULL,      0,    NULL,   0  }
+            };
+
+
+    while (opt != -1) { //getopt creates a data-structure that can be looped through from the argv array to process the parameters and their arguments
+
+        opt = getopt_long(argc, argv, "he:d:", long_options, &option_index);
+
+        switch (opt) {
+            case 'h':
+                printf("Usage: ex1 [OPTION]...[ARG]..\n"
```

Comment:
It might be clearer if you changed `[ARG]` to `[DIRECTORY]` or `[PATH]` or something like that. I think it would just be `[FILE]` on *nix, but not sure about that.